### PR TITLE
KZOO-323: add support for user-defined metadata on call recordings

### PR DESCRIPTION
**Pull request by danielfinke**
- set default `user_metadata` on recordings from account config
- fix mandatory associations in `media_recording:store_info()` map

kazoo5 PR: https://github.com/2600hz/kazoo5/pull/87
kazoo-core PR: https://github.com/2600hz/kazoo-core/pull/536
kazoo-core PR 2: https://github.com/2600hz/kazoo-core/pull/550
kazoo-crossbar PR: https://github.com/2600hz/kazoo-crossbar/pull/386

### CODE REVIEW

File: src/media_recording.erl [hunk 1]
```diff
@@ -39,22 +39,23 @@
...
+                       ,should_store := store_url()
+                       ,retries := non_neg_integer()
+                       ,verb := kz_term:ne_binary()
+                       ,account_id := kz_term:ne_binary()
+                       ,endpoint_id := kz_term:ne_binary()
+                       ,call_id := kz_term:ne_binary()
+                       ,event := kz_call_event:payload()
+                       ,origin := kz_term:ne_binary()
+                       ,pid := pid()
+                       ,user_metadata => kz_json:object()
```
Comment (lazedo) [inline src/media_recording.erl:58]: why this change  ?

Comment (danielfinke) [reply src/media_recording.erl:58]: I tightened up the typing for `store_recording_meta` which was allowing any map, even though it should only allow a map that has these associations (which is why they have been switched to mandatory)
